File: backend/app/schemas/ingest.py
from __future__ import annotations

from pydantic import BaseModel

from app.schemas.db import ClinicDocumentRow


class DocumentListResponse(BaseModel):
    documents: list[ClinicDocumentRow]


class DocumentChunksResponse(BaseModel):
    chunks: list[dict]


class UploadDocumentResponse(BaseModel):
    job_id: str
    status: str
    filename: str


class DeleteDocumentResponse(BaseModel):
    deleted: str

File: frontend/src/types/ingest.ts
/** Ingest API response types (FastAPI /v1/ingest/* + Next.js proxy). */

import type { IngestionJobStatus } from "@/types/database";

export interface IngestUploadResponse {
  job_id: string;
  status: IngestionJobStatus | string;
  filename: string;
}

export interface DeleteDocumentResponse {
  deleted: string;
}

export interface DocumentListResponse {
  documents: import("@/types/clinic-docs").ClinicDocument[];
}

export interface DocumentChunksResponse {
  chunks: import("@/types/clinic-docs").DocumentChunk[];
}
